virtualenv/builders/base.py

```diff
@@ -44,7 +50,35 @@ def __init__(self, python, flavor, system_site_packages=False,
     def check_available(cls, python):
         raise NotImplementedError
 
+    def _get_base_python_bin(self):
+        bindir = json.loads(
+            check_output([
+                self.python,
+                "-c",
+                textwrap.dedent("""
+                import json
+                import os
+                import sys
+                try:
+                    import sysconfig
+                except ImportError:
+                    from distutils import sysconfig
+
+                if (sys.platform.startswith("win") or sys.platform == "cli" and os.name == "nt"):
+                    bindir = getattr(sys, "real_prefix", sys.prefix)
+                else:
+                    bindir = sysconfig.get_config_var("BINDIR")
+
+                print(json.dumps(bindir))
+                """)
+            ]).decode(locale.getpreferredencoding()),
```

Comment:
Will this work on Windows, where getpreferredencoding could well be something like CP850, but Python could be installed in a directory with Unicode characters? Especially in Python 2.5, where per-user installs could be under the user's home directory (and so will be more likely to contain non-ASCII characters). I'd suggest encoding the JSON data in such a way that it is pure-ASCII before sending it to the caller, and then you can decode it back to Unicode on the receiving end.